src/runtime/disco/nccl/nccl.cc

```diff
@@ -140,9 +140,19 @@ struct CCLThreadLocalContext {
   deviceStream_t default_stream = nullptr;
   ncclComm_t comm;
 
-  void Clear() { NCCL_CALL(ncclCommDestroy(comm)); }
+  void Clear() {
+    NCCL_CALL(ncclCommDestroy(comm));
+    if (default_stream != nullptr) {
```

Comment:
IIUC, default_stream is always nullptr

Reply:
given in this case `default_stream` is a member that can be changed later, i think having such code is better. Note it is not the same as the default stream in the CUDA terminology, but in the nccl runtime